src/database/migrations/0042-create-data_provider_roles.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TABLE data_provider_roles (
+  user_keycloak_user_id UUID NOT NULL,
+  data_provider_short_code short_code_t NOT NULL,
+  access_type access_type_t NOT NULL,
+  created_by UUID NOT NULL,
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+  PRIMARY KEY (user_keycloak_user_id, data_provider_short_code, access_type),
+  FOREIGN KEY (created_by) REFERENCES users(keycloak_user_id) ON DELETE CASCADE,
```

Comment:
I find it more readable / intuitive to have single-column foreign keys defined as column constraints rather than table constraints:

```suggestion
  created_by UUID NOT NULL REFERENCES users(keycloak_user_id) ON DELETE CASCADE,
  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
  PRIMARY KEY (user_keycloak_user_id, data_provider_short_code, access_type),
```

Having the FK part of the column definition makes it more obvious to me what the column is, rather than having to look back up at the column once I see the FK table constraint.

Reply:
Sounds good to me